### PR TITLE
Save & Load Button

### PR DESCRIPTION
**Pull request by GoJEGrEEN**
saving and loading player data ...

### CODE REVIEW

**Comment by def-** [comment]
What's this for?

**Comment by FallenKN** [comment]
Seems like it saves the current Playerinfo like Name and Clan into a file so it can be reloaded later, dunno if needed.

**Comment by GoJEGrEEN** [comment]
create backup of your information like skin,name,clan,usecustomcolor,colorbody,colorfeet,countryflag... and restore backup if needed.

**Comment by def-** [comment]
I don't think most people need this. You could also just make a `player.cfg` file with the corresponding entries and execute it with `exec player.cfg`.

**Comment by wp-src** [comment]
WTF . I like this please add this not all players can do what u say @def-

**Comment by def-** [comment]
But the exact same information is already saved and loaded in settings_ddnet.cfg. So I don't understand the purpose of this.

**Comment by GoJEGrEEN** [comment]
yes but I think everyone don't know about your way I mean make player.cfg and exe player.cfg , this is faster and easy you just do it with one click and very usefull for who start playing teeworlds now and don't know anything about this game exec ....

**Comment by wp-src** [comment]
I think he thinks about simple way to doing the job for noob players.

**Comment by nzyuzin** [comment]
To me this feature also seems redundant.

**Comment by GoJEGrEEN** [comment]
the purpose is sometime I want change my colorbody and I forgot the value so I can restore it fast , for example I don't want anyone know me so I try to change my name clan skin colorboy feet .... then I want back on my old setting so I must close client ... but with one click I can do it , this is easy way for newbie

File: src/game/client/components/menus_settings.cpp [hunk 1]
```diff
@@ -301,6 +301,112 @@ void CMenus::RenderSettingsPlayer(CUIRect MainView)
...
+		if(!m_pStorage)
+			return;
+
+		// try to open file
+		IOHANDLE File = m_pStorage->OpenFile("Player_Info.cfg", IOFLAG_WRITE, IStorage::TYPE_SAVE);
+		if(!File)
+			return;
+
+		char aBuf[512];
+		str_format(aBuf, sizeof(aBuf), "%s\r\n%s\r\n%s\r\n%d\r\n%d\r\n%d\r\n%d\r\n%s\r\n%s\r\n%s\r\n%d\r\n%d\r\n%d\r\n%d\r\nsend_info\r\n",
```
Comment (nzyuzin) [inline src/game/client/components/menus_settings.cpp]: Windows style line-endings don't look too good for all of the platforms supported by ddnet.

**Comment by GoJEGrEEN** [comment]
I have not enough experience on other platforms.maybe somewho can fix it.

**Comment by heinrich5991** [comment]
> not all players can do what u say @def-

.

> I think everyone don't know about your way I mean make player.cfg and exe player.cfg , this is faster and easy you just do it with one click and very usefull for who start playing teeworlds now and don't know anything about this game exec ....

.

> about simple way to doing the job for noob players.

.

> but with one click I can do it , this is easy way for newbie

I think we should only *consider* implementing it if someone says they want it. I don't think speculating about a hypothetical "newbie" helps here.

File: src/game/client/components/menus_settings.cpp (hunk 1 above)
Comment (nzyuzin) [inline src/game/client/components/menus_settings.cpp]: There's a io_write_newline() function at https://github.com/ddnet/ddnet/blob/master/src/base/system.h#L251 that handles the choice of line ending for you.

**Comment by Soreu** [comment]
> **GoJEGrEEN commented:**
> the purpose is sometime I want change my colorbody and I forgot the value so I can restore it fast , for example I don't want anyone know me so I try to change my name clan skin colorboy feet .... then I want back on my old setting so I must close client ... but with one click I can do it , this is easy way for newbie

Well, for that most of us use binds...
`bind [key] "player_name \"new nick\"; player_clan \"new clan\"; player_color_body 10092339; player_color_feet 10092339;"`

**Pull request closed without merging**